Add logging error, see #245 & #103

OnnoEbbens opened the pull request.

martinvonk commented:
Nice! Thank you for quickly picking this up.

 Maybe we could be a bit more informative in the error message:
```python
f"Precipitation data not available for station {stn} via the KNMI API."
"If you want the data for this station, please set `use_api` to `False`."
"Now you'll probably receive an empty Obs."
"For more information see issues #245 and #103 on https://github.com/ArtesiaWater/hydropandas/issues"
```

OnnoEbbens commented:
Yeah the weird thing is that for 913 you will get an empty observation and for 967 you get an error. 

I changed it a bit to help hydropandas users more and set `use_api=False` with a warning.

Then I found a small bug when using `fill_missing_obs=True` and `use_api=False` which I fixed

martinvonk commented:
> Yeah the weird thing is that for 913 you will get an empty observation and for 967 you get an error.

Ah, okay that was not clear for me, sorry. This is fine then. :)
